meson.build

```diff
@@ -59,12 +59,7 @@ message('has_exceptions: @0@'.format(has_exceptions))
 
 # GCC or Clang
 if is_gcc or is_clang
-	add_project_arguments(
-		'-march=native',
-		'-fvisibility=hidden',
-		'-fvisibility-inlines-hidden',
-		language: 'cpp'
-	)
+	add_project_arguments('-march=native', language: 'cpp')
```

Comment:
By the way, I think that silently passing `-march=native` to the compiler is not exactly ideal, as it could cause unexpected behaviour for users. It would also prevent every form of packaging, as the build machines' CPUs will always differ from users' ones.

Reply:
Consider that previously the only compiled artefacts were test-related, and you'll understand why that flag is there. Now that you're adding a compiled library target it doesn't make sense, and should instead go in `tests/meson.build`.

Reply:
Ok, I'll fix that too

Reply:
@marzer a lot of flags are currently added in the meson.build files, but none of them were previously used by the consumers of the library since nothing was actually compiled. Are they all intended to be used only for development? If so, leaving them in the `add_project_argiments()` call could create issues (like with `-march=native`). Can I move them all to an array named something like `develop_args` and apply them only in the `tests` directory?

Alright maybe no all of them, for example the flags that enable additional warning are not harmful in any way.

Reply:
>  Are they all intended to be used only for development? 

Yup, that's more-or-less what I was saying above. Since previously there was no compiled code for the end-users, it greatly simplified my life to have them in the top-level project args.

> Can I move them all to an array named something like develop_args and apply them only in the tests directory?

I would like them to apply to the executable targets in `/examples` and `/toml-test`, too, but yes this sounds like a good solution.